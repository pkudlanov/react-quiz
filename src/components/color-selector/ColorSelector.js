import React from 'react';
import PropTypes from 'prop-types';

function ColorSelector({
  fontColor,
  backgroundColor,
  handleFontColorChange,
  handleBackgroundColorChange
}){
  return(
    <>
      <input name="fontColor" type="color" value={fontColor} onChange={handleFontColorChange}/>
      <input name="backgroundColor" type="color" value={backgroundColor} onChange={handleBackgroundColorChange}/>
    </>
  );
}

ColorSelector.propTypes = {
  fontColor: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  handleFontColorChange: PropTypes.func.isRequired,
  handleBackgroundColorChange: PropTypes.func.isRequired
};

export default ColorSelector;
