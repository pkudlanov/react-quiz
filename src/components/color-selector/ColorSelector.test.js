import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('ColorSelector component', () => {
  it('renders ColorSelector', () => {
    const wrapper = shallow(<ColorSelector
      color="green"
      backgroundColor="orange"
      handleColorChange={() => { }}
      handleBackgroundColorChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
