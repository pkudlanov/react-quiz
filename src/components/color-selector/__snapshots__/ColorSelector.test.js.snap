// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector component renders ColorSelector 1`] = `
<Fragment>
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="green"
  />
  <input
    name="backgroundColor"
    onChange={[Function]}
    type="color"
    value="orange"
  />
</Fragment>
`;

exports[`ColorSelector component renders NoteForm 1`] = `
<Fragment>
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="green"
  />
  <input
    name="backgroundColor"
    onChange={[Function]}
    type="color"
    value="orange"
  />
</Fragment>
`;
